tests/dataset_api_tests.py

```diff
@@ -0,0 +1,612 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for Superset"""
+import json
+from typing import List
+
+import prison
+
+from superset import db, security_manager
+from superset.connectors.sqla.models import SqlaTable, TableColumn
+from superset.models.core import Database
+from superset.utils.core import get_example_database
+
+from .base_tests import SupersetTestCase
+
+
+class DatasetApiTests(SupersetTestCase):
+
+    example_create_column_data = {
+        "column_name": "col_exp",
+        "verbose_name": "col verbose name",
+        "description": "col description",
+        "type": "VARCHAR",
+        "groupby": False,
+        "filterable": True,
+        "expression": "col expression",
+        "is_dttm": False,
+        "python_date_format": "epoch_s",
+    }
+
+    @staticmethod
+    def insert_dataset(
+        table_name: str, schema: str, owners: List[int], database: Database
+    ) -> SqlaTable:
+        obj_owners = list()
+        for owner in owners:
+            user = db.session.query(security_manager.user_model).get(owner)
+            obj_owners.append(user)
+        table = SqlaTable(
+            table_name=table_name, schema=schema, owners=obj_owners, database=database
+        )
+        db.session.add(table)
+        db.session.commit()
+        return table
+
+    @staticmethod
+    def insert_dataset_column(
+        table_id: int,
+        column_name: str,
+        verbose_name: str = "",
+        description: str = "",
+        type: str = "",
+        groupby: bool = True,
+        filterable: bool = True,
+        is_dttm: bool = True,
+        expression: str = "",
+        python_date_format: str = "",
```

Comment:
Is there some reason the `str`s aren't `Optional[str] = None`? In the datamodel these seem to be nullable.